### ISSUE TITLE
Require image for ARM64 architecture 

### ISSUE THREAD

**Issue by odidev** [goharbor/harbor-arm#31]
Hi Team, 

I am trying to use **goharbor/harbor-arm** on ARM64 architecture but it seems ARM64 tag is not available in Docker Hub for the same. 

I have raised an issue on **goharbor/harbor** but the member has suggested to open the issue here. 
**Issue Link:** https://github.com/goharbor/harbor/issues/16959 

It seems like you are building it on GHA by self-hosted runner.  

It will be very helpful if the image is released for the ARM64 platform. 
Do you have any plans to release an ARM64 image? 

If required, I am happy to contribute. Can you please provide me with some pointers?

**Comment by Jeremy-boo** [comment]
> Hi Team,
> 
> I am trying to use **goharbor/harbor-arm** on ARM64 architecture but it seems ARM64 tag is not available in Docker Hub for the same.
> 
> I have raised an issue on **goharbor/harbor** but the member has suggested to open the issue here. **Issue Link:** [goharbor/harbor#16959](https://github.com/goharbor/harbor/issues/16959)
> 
> It seems like you are building it on GHA by self-hosted runner.
> 
> It will be very helpful if the image is released for the ARM64 platform. Do you have any plans to release an ARM64 image?
> 
> If required, I am happy to contribute. Can you please provide me with some pointers?

Thank you very much, we are working hard to make the Harbor ARM image. At present, there are some problems in the automated test, so there is no official release. We are solving this problem.

**Comment by odidev** [comment]
@Jeremy-boo Thanks for the reply. 
I was curious to ask what’s your long-term plan for Arm support? Is the long-term plan include keeping **Habor-arm** separate from **Harbor**?

**Comment by Jeremy-boo** [comment]
> 

Our ultimate goal is to hope that arm can be built in the harbor repo, but for now, we will keep harbor-arm separate from harbor for the next period of time

**Comment by odidev** [comment]
Thanks for the quick response.

**Comment by morlay** [comment]
@odidev 

Early access could be found in https://github.com/orgs/octohelm/packages?repo_name=harbor (just newest version)
Without any testing, just image build.

It works well in my cluster (with external redis and pg)

**Comment by jamezrin** [comment]
> @odidev
> 
> Early access could be found in https://github.com/orgs/octohelm/packages?repo_name=harbor (just newest version) Without any testing, just image build.
> 
> It works well in my cluster (with external redis and pg)

How do you use these images? Only the base ones seem to have multiarch support (see [harbor-core](https://github.com/octohelm/harbor/pkgs/container/harbor%2Fharbor-core) vs [harbor-core-base](https://github.com/octohelm/harbor/pkgs/container/harbor%2Fharbor-core-base))

**Comment by morlay** [comment]
@jamezrin Sorry. 
2.5.x changed some envvar in Makefile, I miss it for final push.
https://github.com/goharbor/harbor/compare/release-2.5.0...morlay:release-2.5.0#diff-7747613b3e12f11d61bfe1fd8a6bd18298bc206c697514131a0923aedbc80beeR24

Pls use `v2.5.2` instead (the CI only support compile latest minor version)
<img width="763" alt="image" src="https://user-images.githubusercontent.com/1667873/176805602-ae5b00bd-f7e9-4221-8e46-fcbb13afbf7a.png">

**Comment by jamezrin** [comment]
> @jamezrin Sorry. 2.5.x changed some envvar in Makefile, I miss it for final push. [goharbor/harbor@release-2.5.0...morlay:release-2.5.0diff-7747613b3e12f11d61bfe1fd8a6bd18298bc206c697514131a0923aedbc80beeR24](https://github.com/goharbor/harbor/compare/release-2.5.0...morlay:release-2.5.0diff-7747613b3e12f11d61bfe1fd8a6bd18298bc206c697514131a0923aedbc80beeR24)
> 
> Pls use `v2.5.2` instead (the CI only support compile latest minor version) <img alt="image" width="763" src="https://user-images.githubusercontent.com/1667873/176805602-ae5b00bd-f7e9-4221-8e46-fcbb13afbf7a.png">

Thank you so much, I got harbor running perfectly with that new release. Amazing work!

**Comment by 631068264** [comment]
So how to get arm64 offline installer ?

**Comment by EvanYen** [comment]
> So how to get arm64 offline installer ?

我也需要一个安装脚本

**Comment by kubecto** [comment]
Hope to support as soon as possible, I am still investigating how to deploy arm harbor offline

**Comment by liuxiaodong100** [comment]
I want to arm harbor offline, who can suport me,tks

**Comment by Jeremy-boo** [comment]
> I want to arm harbor offline, who can suport me,tks
We are working on harbor arm construction, any progress will be reflected in this pr, you can pay attention to this pr first
https://github.com/goharbor/harbor/pull/17968

**Comment by sunailong** [comment]
Hi Team,
is anyone keeping solve the issue for make build on arm64 ?

**Comment by renaudrenaud** [comment]
Yes, we clearly want ARM64 solutions, please spend some time on it, we need it!

**Comment by rupinsolanki07** [comment]
Harbor team, any update on ARM builds?

**Comment by Abhinav-26** [comment]
Hey, It has been a long pending request. It would be great to have buildx support for images.

**Comment by renaudrenaud** [comment]
Intel architecture is loosing day after day. ARM and RISC are the future. But for now, we just want harbor for ARM, you can work for RISC later ;)

**Comment by jorikseldeslachts** [comment]
Any progress so far?

**Comment by mrclrchtr** [comment]
Waiting for this...

**Comment by ganyga** [comment]
I'm still waiting...

**Comment by chimisu** [comment]
I'm still waiting...+1

**Comment by feixiaohuijava** [comment]
I'm still waiting...

**Comment by jamezrin** [comment]
If you really need harbor for arm64 you can use https://github.com/octohelm/harbor from @morlay. 

While it is not always up-to-date, it is extremely stable, at least for me. I haven't had any problems in the year and a half I have had it deployed (updated multiple times, currently running 2.9.1)

**Comment by mrclrchtr** [comment]
It looks as if there is not enough development capacity. There is no plan or roadmap to support ARM. They are open to any contribution from the community: https://github.com/goharbor/harbor/issues/20074

**Comment by sakryt** [comment]
Need arm64 offline installer...

**Comment by narunas-vay** [comment]
What are the latest updates? 🤔 Would really appreciate the ARM images 🙏

**Comment by danielfbm** [comment]
Guys, I just added a proposal in the main repo [ARM64 builds? goharbor/harbor#3505](https://github.com/goharbor/harbor/issues/3505)

I believe the way to go forward would be doing the build in parallel right inside the main repo. 

Please help upvote if you agree.

**Comment by michalcukierman** [comment]
As a workaround, one can use Bitnami Charts with  arm support

**Comment by tamalsaha** [comment]
Bitnami charts are not a realistic option anymore https://news.ycombinator.com/item?id=44608856
